.github/workflows/multicoretests.yml

```diff
@@ -0,0 +1,72 @@
+# Run the multicoretests testsuite if PR is labelled with run-multicoretests
+name: Run multicoretests testsuite
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+
+# Restrict the GITHUB_TOKEN
+permissions: {}
+
+# See build.yml
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name == 'pull_request' || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  multicoretests:
+    if: contains(github.event.pull_request.labels.*.name, 'run-multicoretests')
+    runs-on: 'ubuntu-latest'
+    strategy:
+      matrix:
+        include:
+          - profile: dev
+            runparam: b
+          - profile: debug-runtime
+            runparam: b,s=4096
+    steps:
+      - name: Checkout OCaml
+        uses: actions/checkout@v4
+        with:
+          path: ocaml
+          persist-credentials: false
+      - name: Configure, build and install OCaml
+        run: |
+          bash -xe ocaml/tools/ci/actions/multicoretests.sh ocaml
+      - name: Checkout multicoretests
+        uses: actions/checkout@v4
+        with:
+          repository: ocaml-multicore/multicoretests
+          path: multicoretests
+          persist-credentials: false
+      - name: Checkout QCheck
+        uses: actions/checkout@v4
+        with:
+          repository: c-cube/qcheck
+          ref: v0.22
```

Comment:
You could extract the refs to global environment variables earlier in this file. I think it often makes it easier to update the version of the dependencies.

Reply:
You mean pushing the ref into an
```yaml
env:
  QCHECK_REF: v0.22
```
instead? Is that a common pattern?
(I don’t see much difference either way when you want to update the dependencies)

Reply:
yes, I wouldn't say it's common, but I do this sometimes. You do you ;)